argparse_test.go

```diff
@@ -2751,3 +2751,293 @@ func TestCommandHelpSetSnameOnly(t *testing.T) {
+
+	if err := parser.Parse(testArgs1); err != nil {
+		t.Error(err.Error())
+	} else if *intval != 5 {
+		t.Error("Intval did not match expected")
+	} else if *strval != "abc" {
+		t.Error("Strval did not match expected")
+	} else if *floatval != 1.0 {
+		t.Error("Floatval did not match expected")
+	}
+}
+
+func TestCommandPositionalsErr(t *testing.T) {
+	errArgs1 := []string{"posint", "abc", "abc", "1.0"}
+	parser := NewParser("posint", "")
+	_ = parser.IntPositional(nil)
+	_ = parser.StringPositional(nil)
+	_ = parser.FloatPositional(nil)
+
+	if err := parser.Parse(errArgs1); err == nil {
+		t.Error("String argument accepted for integer")
+	} else if err.Error() != "[_positionalArg_posint_1] bad integer value [abc]" {
+		t.Error(err.Error())
+	}
+}
+
+// Just test we don't panic on add
+// Actual I/O during unit tests already covered by TestFileSimple1
+func TestFilePositional(t *testing.T) {
+	parser := NewParser("pos", "")
+	t1 := parser.FilePositional(os.O_RDWR, 0666, nil)
+	t2 := parser.FilePositional(os.O_RDWR, 0666, &Options{Help: "beep!"})
+
+	if t1 == nil {
+		t.Error("File pos was nil")
+	} else if t2 == nil {
+		t.Error("File pos was nil")
+	}
+}
+
+func TestPos1(t *testing.T) {
+	testArgs1 := []string{"pos", "subcommand1", "-i", "2", "abc"}
+	parser := NewParser("pos", "")
+
+	strval := parser.StringPositional(nil)
+	com1 := parser.NewCommand("subcommand1", "beep")
+	intval := com1.Int("i", "integer", nil)
+
+	if err := parser.Parse(testArgs1); err != nil {
+		t.Error(err.Error())
+	} else if *strval != "abc" {
+		t.Error("Strval did not match expected")
+	} else if *intval != 2 {
+		t.Error("intval did not match expected")
+	}
+}
+
+func TestPos2(t *testing.T) {
+	testArgs1 := []string{"pos", "subcommand1", "a123"}
+	parser := NewParser("pos", "")
+
+	strval := parser.StringPositional(nil)
+	com1 := parser.NewCommand("subcommand1", "beep")
+	intval := com1.Int("i", "integer", nil)
+
+	if err := parser.Parse(testArgs1); err != nil {
+		t.Error(err.Error())
+	} else if *strval != "a123" {
+		t.Error("Strval did not match expected")
+	} else if *intval != 0 {
+		t.Error("intval did not match expected")
+	}
+}
+
+func TestPos3(t *testing.T) {
+	testArgs1 := []string{"pos", "subcommand1", "xyz", "--integer", "3"}
+	parser := NewParser("pos", "")
+
+	strval := parser.StringPositional(nil)
+	com1 := parser.NewCommand("subcommand1", "beep")
+	intval := com1.Int("i", "integer", nil)
+
+	if err := parser.Parse(testArgs1); err != nil {
+		t.Error(err.Error())
+	} else if *strval != "xyz" {
+		t.Error("Strval did not match expected")
+	} else if *intval != 3 {
+		t.Error("intval did not match expected")
+	}
+}
+
+func TestPos4(t *testing.T) {
+	testArgs1 := []string{"pos", "abc"}
+	parser := NewParser("pos", "")
+
+	strval := parser.StringPositional(nil)
+	com1 := parser.NewCommand("subcommand1", "beep")
+	intval := com1.Int("i", "integer", nil)
+
+	if err := parser.Parse(testArgs1); err != nil &&
```

Comment:
Fair point, will fix

Reply:
It should fail in that case, I don't see how the current if statement could cause that problem. However I did realize it will pass if the error is nil which is definitely incorrect. Fixed that.